### ISSUE TITLE
[Bug] flash/flicker when opening webpage

### ISSUE THREAD

**Issue by shahrouzz** [react-static/react-static#1561]
### Reporting a bug?

  Provide as much information as you can to reproduce the issue.
  **If the issue is not reproducible, it can't be fixed!**

### Environment

```

react-static 7.4.2

```

### Steps to Reproduce the problem

**Base your steps off of any freshly installed react-static template!**

1.  react-static create react-static-test
2.  yarn stage && yarn serve
3.  open in chrome and run profiling and reload page
4. there will be a frames where the rendering will be cleared and then go back to the index content. this duration is longer depending on how much that needs to be hydrated.

![image](https://user-images.githubusercontent.com/13293917/106111588-0dd73000-6101-11eb-81bd-1f301b771af8.png)

### Expected Behavior

I expect the page to load in chrome without any flickering or graphical if the static html matches the hydrated content.

**Comment by kvet** [comment]
@shahrouzz Seems like I have a solution for the issue.

First, you need manually downgrade `react-universal-component` to 4.0.1. 

Also, take a look at the following thing: https://github.com/faceyspacey/webpack-flush-chunks/pull/82/files. We have applied this fix in order to get proper list of entry chunks by using patch-package

**Comment by shahrouzz** [comment]
@kvet you're absolutely right, by adding
```
"resolutions": {
  "**/react-universal-component": "4.0.1"
}
```
to ```package.json``` I don't see the same problem anymore. thanks!

I am not too familiar with react-static internals and its dependencies so I don't follow what your fix to webpack-flush-chunks does.
Is that a necessary to fix another issue? or can do I need to apply that in order for latest version of ```react-universal-component``` to start working correctly again?

**Comment by kvet** [comment]
@shahrouzz as I understood the second fix is not required for many projects. But if you are using aggressive code splitting it may be needed (like you have many completely different templates and they share some code and css). So, if you find any problem with blinks again, probably you need it. Better to ask project maintainers about it

**Comment by SleeplessByte** [comment]
Yeah. There is a lot of drama with universal component, and I personally have trouble understanding _their_ internals. I'm keen to keeping this issue open until someone smarter than me can figure out a proper fix.

**Comment by AoDev** [comment]
Hello, how do you downgrade react-universal-component? I only have these:

```
  "dependencies": {
    "@reach/router": "^1.3.4",
    "axios": "^0.21.1",
    "react": "^16.14.0",
    "react-dom": "^16.14.0",
    "react-hot-loader": "^4.13.0",
    "react-static": "^7.5.1",
    "react-static-plugin-reach-router": "^7.5.1",
    "react-static-plugin-sitemap": "^7.5.1",
    "react-static-plugin-source-filesystem": "^7.5.1"
  },
```

**Comment by AoDev** [comment]
Ok found how to downgrade. I'm not a yarn user so didn't know about this. Do a yarn install after adding the resolution snippet given by shahrouzz .

**Comment by calinortan** [comment]
Should this fix be applied within react-static? I find it to be a major issue to be honest and it's been a long way until I actually found this thread.

**Comment by SleeplessByte** [comment]
We take PRs @calinortan
